### ISSUE TITLE
smallrye-amqp connector does not respect incoming json content.

### ISSUE THREAD

**Issue by Adelrisk** [quarkusio/quarkus#9217]
**Describe the bug**
The smallrye-amqp connector (in the extension `quarkus-smallrye-reactive-messaging-amqp`) does not seem to respect 'content-type' with the value 'application/json'. It delivers only the raw string to the called method.

**Expected behavior**
In the following method:
```java
    @Incoming("prices")
    @Outgoing("my-data-stream")
    @Broadcast
    public PriceDouble process(Message<JsonObject> priceInUsd) {
```
the metadata declares the content-type as `application/json` and the payload should be an instance of JsonObject.

**Actual behavior**
The payload is a string and the following exception occurs:

```
java.lang.ClassCastException: class java.lang.String cannot be cast to class io.vertx.core.json.JsonObject (java.lang.String is in module java.base of loader 'bootstrap'; io.vertx.core.json.JsonObject is in unnamed module of loader io.quarkus.bootstrap.classloading.QuarkusClassLoader @4b515eab)
        at org.acme.amqp.PriceConverter.process(PriceConverter.java:31)                                                      
        at org.acme.amqp.PriceConverter_ClientProxy.process(PriceConverter_ClientProxy.zig:171)
        at org.acme.amqp.PriceConverter_SmallryeMessagingInvoker_process_fca3565158c7c1217725e8105913919f12c1f8ce.invoke(PriceConverter_SmallryeMessagingInvoker_process_fca3565158c7c1217725e8105913919f12c1f8ce.zig:48)
        at io.smallrye.reactive.messaging.AbstractMediator.invoke(AbstractMediator.java:75)
(and so on)
```

**To Reproduce**

Simply change the example https://github.com/quarkusio/quarkus-quickstarts/tree/master/amqp-quickstart to pass complext objects/classes/JsonObject insead of primitive types:
- create two POJOs with the primitive payload (PrimitiveInteger and PrimitiveDouble)
- update the method signatures to use the POJOs instead of primitives.

**Configuration**

The `amqp-quickstart` for version 1.4.2.Final.

**Additional context**

1. This issue might be related: #8274

2. I've forked the quickstarts to investigate this problem: https://github.com/Adelrisk/quarkus-quickstarts/commits/master.

The first commit https://github.com/Adelrisk/quarkus-quickstarts/commit/42435d2b5d49844eabb60e587826e76780386cfd just swaps out the primitve values for POJOs as described. The exception states that a string can't be cast to (my new) class `PriceInteger`.

The second commit https://github.com/Adelrisk/quarkus-quickstarts/commit/d2cd4dd9aef8bc534482013a2017784e99942cd7 introduces the `Message`, with which I've confirmed that the message content is `application/json`.

3. The prices.html page shows that it is receiving Java addresses from the server instead of a string or JsonObject.

```
The last price is "org.acme.amqp.PriceDouble@5a45a2be" €.
```

**Comment by cescoffier** [comment]
I got it working if both sides are exchanging JsonObject:


```
package org.acme.amqp;

import io.smallrye.mutiny.Multi;
import io.vertx.core.json.JsonObject;
import org.eclipse.microprofile.reactive.messaging.Outgoing;

import javax.enterprise.context.ApplicationScoped;
import java.time.Duration;
import java.util.Random;

@ApplicationScoped
public class PriceGenerator {

    private Random random = new Random();

    @Outgoing("generated-price")
    public Multi<JsonObject> generate() {
        return Multi.createFrom().ticks().every(Duration.ofSeconds(5))
                .map(tick -> random.nextInt(100))
                .map(v -> new JsonObject().put("price", v));
    }

}
```

and:

```
package org.acme.amqp;

import javax.enterprise.context.ApplicationScoped;

import io.vertx.core.json.JsonObject;
import org.eclipse.microprofile.reactive.messaging.Incoming;
import org.eclipse.microprofile.reactive.messaging.Outgoing;

import io.smallrye.reactive.messaging.annotations.Broadcast;

@ApplicationScoped
public class PriceConverter {

    private static final double CONVERSION_RATE = 0.88;

    @Incoming("prices")
    @Outgoing("my-data-stream")
    @Broadcast
    public double process(JsonObject priceInUsd) {
        return priceInUsd.getInteger("price") * CONVERSION_RATE;
    }

}
```

Trying with structures now

**Comment by cescoffier** [comment]
Found the issue, working on a fix.

When encoding an object to JSON, it is passed as AMQP Data instead of AMQP Binary with the content type set as `application/json`.

**Comment by cescoffier** [comment]
Fixed by https://github.com/smallrye/smallrye-reactive-messaging/pull/544.

**Comment by cescoffier** [comment]
So once the fix is merged upstream, this issue would be fixed once:

* smallrye reactive messaging is released (with the fix)
* quarkus gets updated

**Comment by Adelrisk** [comment]
Thanks for your work and the clarifying update in the documentation.

**Comment by serrg1994** [comment]
@cescoffier  I have the same issue with Quarkus 3.1.1


```
  @Incoming("email")
    @Blocking("email-dispatchers-pool", ordered = false)
    fun onEmailNotificationDispatchingRequest(message: JsonObject): List<String> 
```
`    
java.lang.String cannot be cast to class io.vertx.core.json.JsonObject (java.lang.String is in module java.base of loader 'bootstrap'; io.vertx.core.json.JsonObject is in unnamed module of loader io.quarkus.bootstrap.classloading.QuarkusClassLoader`


I'm sending message from console running as a Dev service

**Comment by cescoffier** [comment]
Please open another issue and attach a standalone reproducer.
